### ISSUE TITLE
[BUG] SMAC3 not compatible with sklearn 1.9 due to changes in their tree models (RF)

### ISSUE THREAD

**Issue by thomasATbayer** [automl/SMAC3#1336]
There have been changes to the tree  models in sklearn 1.9, making it incompatible with smac3 ("cannot import name 'DTYPE' from 'sklearn.tree._tree')

#### Description
I tried to run smac3 with sklearn 1.9

#### Steps/Code to Reproduce

from ConfigSpace import Configuration, ConfigurationSpace

import numpy as np
from smac import HyperparameterOptimizationFacade, Scenario
from sklearn import datasets
from sklearn.svm import SVC
from sklearn.model_selection import cross_val_score

iris = datasets.load_iris()

def train(config: Configuration, seed: int = 0) -> float:
    classifier = SVC(C=config["C"], random_state=seed)
    scores = cross_val_score(classifier, iris.data, iris.target, cv=5)
    return 1 - np.mean(scores)

configspace = ConfigurationSpace({"C": (0.100, 1000.0)})

scenario = Scenario(configspace, deterministic=True, n_trials=200)

smac = HyperparameterOptimizationFacade(scenario, train)
incumbent = smac.optimize()

#### Expected Results
No error message

#### Actual Results
{
	"name": "ImportError",
	"message": "cannot import name 'DTYPE' from 'sklearn.tree._tree' (/home/sagemaker-user/MotherML/.venv/lib/python3.12/site-packages/sklearn/tree/_tree.cpython-312-x86_64-linux-gnu.so)",
	"stack": "\u001b[31m---------------------------------------------------------------------------

#### Versions
smac==2.4.0

**Comment by angrinord** [comment]
Issue fixed in main #1325 and development #1346
